scvi/dataset/synthetic.py

```diff
@@ -42,3 +42,113 @@ def __init__(self, mu=4.0, theta=2.0, dropout=0.7, save_path='data/'):
         super().__init__(
             *GeneExpressionDataset.get_attributes_from_matrix(X_train, labels=labels_simlr)
         )
+
+
+class SyntheticDatasetCorrelated(GeneExpressionDataset):
+    def __init__(self, n_cells_cluster=100, n_clusters=5,
+                 n_genes_high=15, n_genes_total=50,
+                 weight_high=0.1, weight_low=0.01,
+                 lam_0=100., p_dropout=0.2,
+                 n_batches=1, n_labels=1, mode='mixed', ratio_genes_zi=0.5):
+        """
+
+        :param n_cells_cluster: Number of cells in each cluster
+        :param n_clusters: Number of cell clusters
+        :param n_genes_high: Number of highly expressed genes for each cluster.
+        Other genes are still expressed but at a lower expression
+        The level expression for highly expressed genes follow the same distribution
+
+        :param n_genes_total: Number of genes in total
+        :param weight_high: Level weight for highly expressed genes
+        :param weight_low: Level weight for lowly expressed genes
+        :param lam_0: Proportionality in the Poisson distributions parameter
+        :param p_dropout: Probability of dropout (only makes sense if mode='zi'
+        or mode='mixed'
+        :param n_batches: (useless for now) Number of batches in dataset
+        :param n_labels: (useless for now) Number of labels
+        :param mode: Should be either
+            - 'nb': mode without zero inflation
+            - 'zi': mode including zero inflation for ALL genes
+            -'mixed': mode in which SOME genes are Zero-Inflated while some other are not
+            The ratio_genes_zi parameter allow to control the proportion of genes that are going
+            to be ZI
+        :param ratio_genes_zi: Proportion of ZI genes
+        """
+
+        assert mode in ['mixed', 'zi', 'nb'], 'Mode {} not recognized'.format(mode)
+        np.random.seed(0)
+
+        if n_genes_total % n_clusters > 0:
+            print("Warning, clusters have inequal sizes")
+
+        if n_genes_high > (n_genes_total // n_clusters):
+            print("Overlap of", n_genes_high - (n_genes_total // n_clusters), "genes")
+
+        # Generate data before dropout
+        batch_size = n_cells_cluster * n_clusters
+        data = np.ones((n_batches, batch_size, n_genes_total))
```

Comment:
Better to give this a more descriptive name than "data" (all variables store data). You might call it the `expression_mat` (expression matrix), or `count_mat`, or something like that.

Reply:
OK we change that immediately, thanks!